### ISSUE TITLE
DPL-141 When scanning the “LTHR-384 Pool XP tube” and the “LB Lib Pool Norm” tube we would like to verify the barcodes on the tubes are from the same parent samples prior to manual transfer of the liquid within [Heron] (M)

### ISSUE THREAD

**Issue by stevieing** [sanger/limber#799]
**User story**
When scanning the “LTHR-384 Pool XP tube” and the “LB Lib Pool Norm” tube we would like to verify the barcodes on the tubes are from the same parent samples prior to manual transfer of the liquid within.

Currently swaps are occasionally being made during the normalisation step, even though the team are only normalising one pool on a rack at a time.

Potentially this is due to the wrong barcode label being selected to stick to the norm tube.

**Who are the primary contacts for this story**
Naomi / Sarah / Steve

**Acceptance criteria**
To be considered successful the solution must allow:
- [ ] a simple “scan both tubes and verify for a single pool” page
- [ ] Nice to have: Once tubes verified, display the source molarity, volume to transfer and buffer to add. Do not do this before tubes are verified, to avoid people transferring the wrong volume

**Dependencies**
This story is blocked by the following dependencies:
- #923 

References
This story has a non-blocking relationship with:
- #<issue_no.>
- sanger/<repo>#<issue_no.>

**Additional context**

**Comment by neilsycamore** [comment]
Swap incident last week, the XP tube barcodes were one digit diff
 NT1682752, NT1682751 tho why they ignore the checksum? NT1682752N, NT1682751M

**Comment by JamesGlover** [comment]
Earlier comment probably wrong, as we're actually generating robot driver files for this step in DPL-112
https://github.com/sanger/limber/issues/795
This suggests we'll be transferring between multiple tubes at the same time. We may also need to validate locations, dependent on the CSV. (Or else CSV will be generated by the robot itself)

**Comment by JamesGlover** [comment]
Little confused, as Naomi says this *is* a manual transfer, so now I'm not sure why we are being asked to generate a CSV.

**Comment by JamesGlover** [comment]
So this story *is* as described.

**Comment by sdjmchattie** [comment]
After some correspondence with Naomi P, we have a pretty clear idea of what this should do:

- Right after the creation of a new LB Lib Pool Norm tube, the tube should offer an option to verify or pair it with the LTHR-384 Pool XP tube it was generated by.
- The verify page will accept a scan of an LTHR-384 Pool XP tube and an LB Lib Pool Norm tube.  The page checks that they have the same ancestry before setting the tube up for Manual Transfer.
- Only after verification should the tube show the transfer volumes alongside the QC results.
- The tube can then be manually transferred as normal.

**Comment by sdjmchattie** [comment]
After careful consideration and the complexity of changing the way the Norm Tubes work, we have come up with a workflow we can definitely implement.  See attached.  This has been validated as a good solution for their needs by Sara S and Naomi P.

[Tube Validation Workflow.pdf](https://github.com/sanger/limber/files/7608565/Tube.Validation.Workflow.pdf)
